services/outputhost/messagecache.go

```diff
@@ -537,18 +548,18 @@ func (msgCache *cgMsgCache) utilRenewCredits() {
 	}
 }
 
-func (msgCache *cgMsgCache) addTimer(delaySeconds int, id AckID) common.UnixNanoTime {
+func (msgCache *cgMsgCache) addTimer(delayMilliSeconds int32, id AckID) common.UnixNanoTime {
```

Comment:
probably better to take delay of type "time.Duration" instead of "int32" milliseconds, etc. and at the call-site you could just multiply by "time.Second", etc; and re-define "nackRedeliveryDelayInMillisecond" as simply "nackRedeliveryDelay" (that can be expressed in time.Duration).